### ISSUE TITLE
Problem with Nullable Types

### ISSUE THREAD

**Issue by escamoteur** [joanpablo/reactive_forms#146]
Hi,

thanks for this amazing package that I currently use in my little side project. Unfortunately I ran into a problem with a Datetime field that is allowed to be Null in my case.

Instead of long describing the problem, please watch here as it happened during a live coding session https://youtu.be/hjgMgffk-5A?t=3727

**Comment by joanpablo** [comment]
Hi @escamoteur,

I haven't seen yet the video completely, but as far as I have seen until now I saw a misunderstood in the definition of a FormControl.

In the video you defined a control as **FormControl<DateTime?>** instead of just **FormControl&lt;DateTime&gt;**. L:et me explain my self:

The nature of a FormControl is to be nullable, so you define the data type as usually **T** but all properties related to the central such as: **FormControl.value** and many others will assume automatically that you are dealing with a **T?** data type. So no need to define a **DateTime?**.

I know it can be a bit confusing now that we have null-safety in Flutter, and the first thing that comes into one's mind is to declare a FormControl data type as **?** to accept nulls, but control by default will assume that a value will be always nullable by default, and you can use Validators like **Validators.required** to force the not nullable in UI.

Anyway, you have made me rethink again, and ask again myself if Reactive Forms should assume that values can be nullable or let that the developer decide it when defining the data type.

Thanks for the issue, I will continue watching the video and let you know my final thougs.

**Comment by escamoteur** [comment]
Hmm interesting. will change it back and try to understand again what the problem was.

**Comment by joanpablo** [comment]
I agree with you that this should be more explicit in the documentation, I definitely need to make something because now that we have nullable types in Flutter I'm guessing this is going to confuse many people because the package is not clear enough about this.

**Comment by escamoteur** [comment]
Just tried and it works fine. I really guess it was that it felt wrong to not use a `DateTime?` if I expect that the field can contain a null value.

**Comment by joanpablo** [comment]
I will rethink again all this about nullable types, meanwhile thanks a lot, and don't hesitate to open a new issue if you have any other questions.

**Comment by joanpablo** [comment]
You can define a FormControl as double and use a ReactiveTextField with an [TextInputFormatter](https://api.flutter.dev/flutter/services/TextInputFormatter-class.html) that validates the input and formats correctly the currency value. We actually have a Validators.number but it will not be useful to validate floating point because it just validates int (I need to create a Validators.floatingNumber) but if you use a TextInputFormatter you will not need the validator (or you can create your own custom validator meanwhile)

This [issue](https://github.com/joanpablo/reactive_forms/issues/141) is maybe related to your question.
